cypress-tests/cypress/e2e/dashboardV2.cy.ts

```diff
@@ -0,0 +1,77 @@
+import { TIMEOUTS } from "../../config";
+import { getRandomString, validateLogin } from "../support/commands/general";
+import { generatorProjectName } from "../support/commands/projects";
+
+const username = Cypress.env("TEST_USERNAME");
+
+const projectTestConfig = {
+  projectAlreadyExists: false,
+  projectName: generatorProjectName("dashboardV2"),
+};
+
+export type ProjectIdentifier = {
+  slug: string;
+  namespace?: string;
+};
+
+const projectIdentifier = {
+  slug: projectTestConfig.projectName
+};
+
+const prefixProjectTitle = "My Renku Project";
+
+const sessionId = ["dashboardV2", getRandomString()];
+
+describe("Basic Dashboard functionality", () => {
+  let userNamespace: string | null = null;
+
+  after(() => {
+    if (!projectTestConfig.projectAlreadyExists)
+      cy.deleteProjectFromAPIV2({ slug: projectIdentifier.slug, namespace: userNamespace });
+  });
+
+  beforeEach(() => {
+    // Restore the session
+    cy.session(
+      sessionId,
+      () => {
+        cy.robustLogin();
+      },
+      validateLogin
+    );
+    cy.getUserNamespaceAPIV2().then((namespace) => {
+      if (namespace) {
+        userNamespace = namespace;
+        cy.createProjectIfMissingAPIV2({
+          visibility: "private",
+          name: `${prefixProjectTitle} ${projectIdentifier.slug}`,
+          namespace,
+          slug: projectIdentifier.slug,
+        });
+      } else {
+        cy.log('No user namespace found, project cannot be created.');
+      }
+    });
+  });
+
+  it("Can find own project", () => {
+    cy.visit("v2");
+    cy.getDataCy("dashboard-project-list").find("a").should("have.length.at.least", 1);
+    cy.getDataCy("dashboard-project-list").find("a").should("contain.text", `${prefixProjectTitle} ${projectIdentifier.slug}`);
+    cy.getDataCy("dashboard-project-list").find("a").should("contain.text", projectIdentifier.slug);
+
+    //check if the project also exist in the search
+    cy.wait(TIMEOUTS.minimal); // eslint-disable-line cypress/no-unnecessary-waiting
+    cy.getDataCy("view-my-projects-btn").click();
+    cy.getDataCy("search-card").should("have.length.at.least", 1);
+    cy.getDataCy("search-card").should("contain.text", `${prefixProjectTitle} ${projectIdentifier.slug}`);
+
+    // logout and search for the project and log back in
+    cy.logout();
+    cy.visit("v2");
+    cy.getDataCy("projects-container").contains("No 2.0 projects.");
+    cy.getDataCy("view-other-projects-btn").click();
+    cy.getDataCy("search-results").should("not.contain.text", `${prefixProjectTitle} ${projectIdentifier.slug}`);
+    cy.robustLogin();
```

Comment:
suggestion: I think the three sections in this single `it` clause should be 3 separate `it` clauses. Something like:
- `it("Can find a private project in the dashboard")`
- `it("Can find a private project in the search")`
- `it("Cannot find a private project when logged out")`

And the describe section above should be something like `Basic private project functionality` rather than `Basic Dashboard functionality`. Because only one set of statements in this test (the first three) test anything about the dashboard.